providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -370,6 +370,17 @@ static int ml_kem_key_fromdata(ML_KEM_KEY *key,
         return 0;
     }
 
+    /* Check any explicit public key against embedded value in private key */
+    if (publen > 0 && prvlen > 0) {
+        puboff = prvlen - ML_KEM_RANDOM_BYTES - ML_KEM_PKHASH_BYTES - publen;
```

Comment:
Doesnt look like we have a size of the DKpke stored in the vinfo which is (768 * k) which is the puboff..

A comment here might be nice e.g.
point to the ek offset in the DKpke||ek||H(ek)||z